packages/preview/algorithmic-0.1.0/README.md

```diff
@@ -0,0 +1,135 @@
+<!--
+SPDX-FileCopyrightText: 2023 Jade Lovelace
+
+SPDX-License-Identifier: MIT
+-->
+
+# typst-algorithmic
+
+This is a package inspired by the LaTeX [`algorithmicx`][algorithmicx] package
+for Typst. It's useful for writing pseudocode and typesetting it all nicely.
+
+[algorithmicx]: https://ctan.org/pkg/algorithmicx
+
+![screenshot of the typst-algorithmic output, showing line numbers, automatic
+indentation, bolded keywords, and such](./docs/assets/demo-rendered.png)
+
+Example:
+
+```typst
+#import "algorithmic.typ"
+#let algorithm = algorithmic.algorithm
```

Comment:
This would perhaps be a bit cleaner?
```suggestion
#import algorithmic: algorithm
```

Reply:
I don't think this works; I use an import below to grab * from the algorithmic item defined here, in the DSL scope.

Reply:
I mean just replacing the leet line and keeping the `#import "algorithmic.typ"`. `#let algorithm = algorithmic.algorithm` and `#import algorithmic: algorithm` should be pretty much equivalent.

Reply:
```
#import "algorithmic.typ": algorithm

#algorithm({
    import algorithmic: *
    Cmt[blah]
})
```

yields:

```
watching test.typ
writing to test.pdf

[14:26:25] compiled with errors

error: unknown variable: algorithmic
  ┌─ /test.typ:4:11
  │
4 │     import algorithmic: *
  │            ^^^^^^^^^^^
```

Reply:
I mean
```
#import "algorithmic.typ"
#import algorithmic: algorithm

#algorithm({
    import algorithmic: *
    Cmt[blah]
})
```

Reply:
It's also not a big deal at all, I just think it reads a little nicer.

Reply:
fixed